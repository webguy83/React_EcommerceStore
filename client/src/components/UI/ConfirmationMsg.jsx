import React from "react";
import styled, { css } from "styled-components/macro";

const backgroundColour = css`
  background-color: ${({ status }) => {
    switch (status) {
      case "success":
        return "var(--success);";
      case "fail":
        return "var(--fail);";
      default:
        return "var(--warning);";
    }
  }};
`;

const colourTrans = css`
  transition: all 2s;
`;

const ConfirmMsg = styled.div`
  ${backgroundColour}
  overflow: hidden;
  ${colourTrans}
`;

const StatusMsg = styled.p`
  font-size: 1.5rem;
  text-align: center;
`;

const ConfirmationMsg = ({ status, statusMessages }) => {
  return (
    <ConfirmMsg data-testid="container" status={status}>
      <StatusMsg data-testid="statusMsg">{statusMessages[status]}</StatusMsg>
    </ConfirmMsg>
  );
};

export default ConfirmationMsg;
