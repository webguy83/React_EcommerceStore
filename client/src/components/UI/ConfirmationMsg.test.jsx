import React from "react";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";
import ConfirmationMsg from "./ConfirmationMsg";

afterEach(cleanup);

const defaultStatusMessage = {
  success: "You have submitted without error!",
  fail: "You have failed!",
  sending: "Loading..."
};

describe("confirmation box - success messages", () => {
  it("should render without crashing", () => {
    const { asFragment } = render(
      <ConfirmationMsg status="success" statusMessages={defaultStatusMessage} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render success message when successful", () => {
    const { getByTestId } = render(
      <ConfirmationMsg status="success" statusMessages={defaultStatusMessage} />
    );
    expect(getByTestId("statusMsg")).toHaveTextContent(
      /^You have submitted without error!$/
    );
  });

  it("should render fail message when failed", () => {
    const { getByTestId } = render(
      <ConfirmationMsg status="fail" statusMessages={defaultStatusMessage} />
    );
    expect(getByTestId("statusMsg")).toHaveTextContent(/^You have failed!$/);
  });

  it("should render loading message when loading", () => {
    const { getByTestId } = render(
      <ConfirmationMsg status="sending" statusMessages={defaultStatusMessage} />
    );
    expect(getByTestId("statusMsg")).toHaveTextContent(/^Loading...$/);
  });
});

describe("confirmation box - status background colour", () => {
  it("should set background colour of status bar to green when success", () => {
    const { getByTestId } = render(
      <ConfirmationMsg status="success" statusMessages={defaultStatusMessage} />
    );
    expect(getByTestId("container")).toHaveStyleRule(
      "background-color",
      "var(--success)"
    );
  });

  it("should set background colour of status bar to red when failed", () => {
    const { getByTestId } = render(
      <ConfirmationMsg status="fail" statusMessages={defaultStatusMessage} />
    );
    expect(getByTestId("container")).toHaveStyleRule(
      "background-color",
      "var(--fail)"
    );
  });

  it("should set background colour of status bar to yellow when loading", () => {
    const { getByTestId } = render(
      <ConfirmationMsg status="sending" statusMessages={defaultStatusMessage} />
    );
    expect(getByTestId("container")).toHaveStyleRule(
      "background-color",
      "var(--warning)"
    );
  });
});
