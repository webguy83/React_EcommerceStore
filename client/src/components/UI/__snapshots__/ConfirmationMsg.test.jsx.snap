// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`confirmation box - success messages should render without crashing 1`] = `
<DocumentFragment>
  .c0 {
  background-color: var(--success);
  overflow: hidden;
  -webkit-transition: all 2s;
  transition: all 2s;
}

.c1 {
  font-size: 1.5rem;
  text-align: center;
}

<div
    class="c0"
    data-testid="container"
  >
    <p
      class="c1"
      data-testid="statusMsg"
    >
      You have submitted without error!
    </p>
  </div>
</DocumentFragment>
`;
